### ISSUE TITLE
Raise exceptions if non-physical parameters are defined in forces

### ISSUE THREAD

**Issue by jchodera** [openmm/openmm#3185]
The Open Force Field folks (@dotsdl @j-wags) did some great detective work to identify that their automated force field parameterization tool accidentally defined a reference angle for `HarmonicAngleForce` that was > 180 degrees. These non-physical parameters don't generally make sense, and it would be much easier if we could throw exceptions when users try to define values outside the allowed domain to make it easier to identify programming or logic errors.

For example, we could check for
* Angles or torsions outside defined domains
* LJ sigma or epsilon values that are negative
* GB radii that are negative
* etc.

**Comment by peastman** [comment]
Good idea.  We do need to be careful that we don't limit our applications.  For example, a coarse grained model or a toy model might have parameters that wouldn't be appropriate in a realistic atomic model.  But there are some parameters that really just don't make sense, like a bond length or angle being negative.

**Comment by Lnaden** [comment]
I like the idea of defining bounds a few physical parameters such as negative bond lengths as @peastman suggested. Would it be possible to define some of these boundaries then permit an override kwarg that lets the user say "Yes, I know what I'm doing and violating the defined boundaries"? That may be adding too much complexity but it would provide minimal disruption to people who do want to break the boundaries (such as the CG example) without having to wait for a new release.
